boa/src/exec/conditional/mod.rs

```diff
@@ -0,0 +1,19 @@
+use super::{Executable, Interpreter};
+use crate::{
+    builtins::{ResultValue, Value},
+    syntax::ast::node::If,
+};
+use std::borrow::Borrow;
+
+impl Executable for If {
+    fn run(&self, interpreter: &mut Interpreter) -> ResultValue {
+        Ok(if self.cond().run(interpreter)?.borrow().is_true() {
+            self.body().run(interpreter)?
+        } else {
+            match self.else_node() {
+                Some(ref else_e) => else_e.run(interpreter)?,
+                None => Value::undefined(),
+            }
+        })
```

Comment:
Could this be reduced to an if let at the level of the else?
```suggestion
        } else if let Some(ref else_e) = self.else_node() {
            else_e.run(interpreter)?
        } else {
        	Value::undefined()
        })
```

Reply:
Done :)